erigon-lib/state/domain.go

```diff
@@ -1613,12 +1624,22 @@ func (dt *DomainRoTx) statelessGetter(i int) *seg.Reader {
 	}
 	r := dt.getters[i]
 	if r == nil {
-		r = seg.NewReader(dt.files[i].src.decompressor.MakeGetter(), dt.d.compression)
+		r = dt.newReader(dt.files[i].src.decompressor.MakeGetter())
 		dt.getters[i] = r
 	}
 	return r
 }
 
+func (dt *DomainRoTx) newReader(getter *seg.Getter) *seg.Reader {
+	compression := dt.d.compression
+	if dt.name == kv.CommitmentDomain { // alow disable compression in future
+		if compression != seg.DetectCompressType(getter) {
+			compression = dt.d.compression
```

Comment:
similar as the previous comment, but here you are just reassigning dt.d.compression like 3 lines before?